plugins/registry.js

```diff
@@ -49,6 +49,39 @@ const resolveReactElement = (element) => {
     return element;
 };
 
+class SubMenuGroup {
+    constructor(pluginId, parentId, depth) {
+        this.pluginId = pluginId;
+        this.parentId = parentId;
+        this.depth = depth;
+        this.registerMenuItem = this.registerMenuItem.bind(this);
+        this.subMenuGroup = [];
+    }
+    registerMenuItem(item, action, filter) {
+        this.id = generateId();
+
+        store.dispatch({
+            type: ActionTypes.RECEIVED_PLUGIN_MENUITEM,
+            name: 'PostDropdownMenu',
+            data: {
+                id: this.id,
+                parentId: this.parentId,
+                pluginId: this.pluginId,
+                text: resolveReactElement(item),
+                action,
+                filter,
+            },
+        });
+
+        this.subMenuGroup.push(
```

Comment:
I don't think this is the right approach: we're effectively mutating the Redux store here. Should we just dispatch an event here, and have the reducer insert the new menu item into the data structure?

Reply:
The array push was added to maintain an object linkage to pass along pluginId and parentId.  This is likely best rewritten as a higher-order function as you describe below.